docs/executor/kubernetes.rst

```diff
@@ -44,15 +44,21 @@ KubernetesExecutor Architecture
 The KubernetesExecutor runs as a process in the Scheduler that only requires access to the Kubernetes API (it does *not* need to run inside of a Kubernetes cluster). The KubernetesExecutor requires a non-sqlite database in the backend, but there are no external brokers or persistent workers needed.
 For these reasons, we recommend the KubernetesExecutor for deployments have long periods of dormancy between DAG execution.
 
+When a DAG submits a task, the KubernetesExecutor requests a worker pod from the Kubernetes API. The worker pod then runs the task, reports the result, and terminates.
 
-.. image:: ../img/k8s-0-worker.jpeg
 
+.. image:: ../img/arch-diag-kubernetes.png
```

Comment:
I really appreciate the addition, but I think we should know how to reproduce the image in the future if we have to update it :). I really love things like mermaid - that can generate the image from markdown-ish text description and have everyone contribute to it easily. Maybe we can re-create the graphs with it . See  #10380 - mermaid has the capability of generating nice diagrams from the textual description which I think is crucial to get images that we can update in the future as a community effort.

Reply:
The source has been uploaded to https://cwiki.apache.org/confluence/download/attachments/158872928/arch-diag-kubernetes.drawio?api=v2 under https://cwiki.apache.org/confluence/display/AIRFLOW/Drawio+Diagrams